File: su/src/test/java/og/sqlbuilder/sumo/TestWordResolver.java
package og.sqlbuilder.sumo;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.sqlbuilder.su.KBLoader;
import org.sqlbuilder.su.SuWordResolver;

import java.io.IOException;

public class TestWordResolver
{
	static SuWordResolver resolver;

	@BeforeAll
	public static void init()
	{
		final String ser = System.getenv().get("SUWORDRESOLVER");
		resolver = new SuWordResolver(ser);
	}

	@Test
	public void testResolver()
	{
		var id = resolver.apply("airport");
		System.out.println(id);
	}

	public static void main(String[] args) throws IOException
	{
		new KBLoader().load();
		init();
		new TestWordResolver().testResolver();
	}
}

File: su/src/test/java/og/sqlbuilder/sumo/TestWordResolver.kt
package og.sqlbuilder.sumo

import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.sqlbuilder.su.KBLoader
import org.sqlbuilder.su.SuWordResolver
import java.io.IOException

class TestWordResolver {

    @Test
    fun testResolver() {
        val id: Int? = resolver.invoke("airport")
        println(id)
    }

    companion object {

        lateinit var resolver: SuWordResolver

        @JvmStatic
        @BeforeAll
        fun init() {
            val ser: String = System.getenv()["SUWORDRESOLVER"]!!
            resolver = SuWordResolver(ser)
        }

        @JvmStatic
        @Throws(IOException::class)
        fun main(args: Array<String>) {
            KBLoader().load()
            init()
            TestWordResolver().testResolver()
        }
    }
}
